src/app_model/registries/_keybindings_reg.py

```diff
@@ -23,8 +29,28 @@ class _RegisteredKeyBinding(NamedTuple):
     keybinding: KeyBinding  # the keycode to bind to
     command_id: str  # the command to run
     weight: int  # the weight of the binding, for prioritization
+    source: KeyBindingSource  # who defined the binding, for prioritization
     when: expressions.Expr | None = None  # condition to enable keybinding
 
+    def __gt__(self, other: object) -> bool:
+        if not isinstance(other, _RegisteredKeyBinding):
+            return NotImplemented
+        return self.source.value > other.source.value or (
+            self.source == other.source and self.weight > other.weight
+        )
+
+    def __lt__(self, other: object) -> bool:
+        if not isinstance(other, _RegisteredKeyBinding):
+            return NotImplemented
+        return self.source.value < other.source.value or (
+            self.source == other.source and self.weight < other.weight
+        )
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, _RegisteredKeyBinding):
+            return NotImplemented
+        return self.source.value == other.source.value and self.weight == other.weight
```

Comment:
after making `KeyBindingSource` an `IntEnum`:

```suggestion
    def __gt__(self, other: object) -> bool:
        if not isinstance(other, _RegisteredKeyBinding):
            return NotImplemented
        return (self.source, self.weight) > (other.source, other.weight)

    def __lt__(self, other: object) -> bool:
        if not isinstance(other, _RegisteredKeyBinding):
            return NotImplemented
        return (self.source, self.weight) < (other.source, other.weight)

    def __eq__(self, other: object) -> bool:
        if not isinstance(other, _RegisteredKeyBinding):
            return NotImplemented
        return (self.source, self.weight) == (other.source, other.weight)
```